### PR TITLE
[Improvement-11773][api] Optimize the log printing of the api module according…

### LINKED ISSUE #11773: [Improvement][api] Optimize the log printing of the api module according to the log specification.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

The log printing in the current api module is not very standardized.
According to the log specification in pr [#11484](https://github.com/apache/dolphinscheduler/pull/11484), I will improve the log printing of the api module.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by guowei-su**
## Purpose of the pull request
This pull request optimizes the log printing of the api module.

## Brief change log
- Add or modify log of the files in api/service/impl/, etc.
- close #11773
- doc [[Improvement-11481][docs] Add log specification document to contribution guidelines. #11484](https://github.com/apache/dolphinscheduler/pull/11484)

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by SbloodyS — COMMENTED** [review]
Please associate the issue and resolves conflicts.

**Comment by guowei-su** [comment]
issue [#11773](https://github.com/apache/dolphinscheduler/issues/11773)

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java [hunk 1]
```diff
@@ -107,6 +107,8 @@ public boolean resourcePermissionCheck(Object authorizationType, Object[] needCh
             Set<?> originResSet = new HashSet<>(Arrays.asList(needChecks));
             Set<?> ownResSets = RESOURCE_LIST_MAP.get(authorizationType).listAuthorizedResource(userId, logger);
             originResSet.removeAll(ownResSets);
+            if (!originResSet.isEmpty())
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java]:
```suggestion
            if (CollectionUtils.isNotEmpty(originResSet)) {
                 logger.warn("User does not have resource permission {} on associated resources, userId:{}", authorizationType, userId);
            }
```

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TenantServiceImpl.java [hunk 2]
```diff
@@ -229,9 +232,14 @@ public Map<String, Object> updateTenant(User loginUser, int id, String tenantCod
         if (!Objects.equals(existsTenant.getTenantCode(), updateTenant.getTenantCode()) && PropertyUtils.getResUploadStartupState()) {
             storageOperate.createTenantDirIfNotExists(tenantCode);
         }
-        tenantMapper.updateById(updateTenant);
-
-        putMsg(result, Status.SUCCESS);
+        int update = tenantMapper.updateById(updateTenant);
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TenantServiceImpl.java:235]: Need to update the UT case.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TenantServiceImpl.java [hunk 3]
```diff
@@ -278,10 +290,16 @@ public Map<String, Object> deleteTenantById(User loginUser, int id) throws Excep
           storageOperate.deleteTenant(tenant.getTenantCode());
         }
 
-        tenantMapper.deleteById(id);
-        processInstanceMapper.updateProcessInstanceByTenantId(id, -1);
+        int delete = tenantMapper.deleteById(id);
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TenantServiceImpl.java:293]: Need to update the UT case.

**Comment by caishunfeng** [comment]
Please add a doc of `How to print logs better`.

**Comment by guowei-su** [comment]
> Please add a doc of `How to print logs better`.

The doc refs this doc pr: [Add log specification document to contribution guidelines. ](https://github.com/apache/dolphinscheduler/pull/11484)

**Comment by guowei-su** [comment]
PLAT @caishunfeng

**Review by brave-lee — DISMISSED** [review]
+1

**Review by brave-lee — APPROVED** [review]
+1

**Comment by zhongjiajie** [comment]
I do not think this is a good idea, we should directly throw error instead of print log, and actually I already throw error in #11912

**Comment by zhongjiajie** [comment]
> I do not think this is a good idea, we should directly throw error instead of print log, and actually I already throw error in #11912

We already have `ExceptionHandler`accept `ServiceException` and package `ServiceException` code and message into `Result` object, and it uses in the controller layer https://github.com/apache/dolphinscheduler/blob/ae6aa53f963d2ea02ab435db1e5548da23e3f882/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/exceptions/ApiExceptionHandler.java#L39-L43
so when we throw an exception in the service layer, we will catch it in the controller layer and return the new `Result` object. And we already have exists method in https://github.com/apache/dolphinscheduler/blob/d3a77c68e62b0c471e25cd6a4bfbd2c561dd2023/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java#L657-L659. 

when exception throw in L658 https://github.com/apache/dolphinscheduler/blob/d3a77c68e62b0c471e25cd6a4bfbd2c561dd2023/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java#L658
controller will catch it and package into `Result(code=Status.PROCESS_INSTANCE_STATE_OPERATION_ERROR.getCode(), message=Status.PROCESS_INSTANCE_STATE_OPERATION_ERROR.getMsg())`

**Comment by guowei-su** [comment]
> > I do not think this is a good idea, we should directly throw error instead of print log, and actually I already throw error in #11912
> 
> We already have `ExceptionHandler`accept `ServiceException` and package `ServiceException` code and message into `Result` object, and it uses in the controller layer
> 
> https://github.com/apache/dolphinscheduler/blob/ae6aa53f963d2ea02ab435db1e5548da23e3f882/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/exceptions/ApiExceptionHandler.java#L39-L43
> 
> 
> so when we throw an exception in the service layer, we will catch it in the controller layer and return the new `Result` object. And we already have exists method in
> https://github.com/apache/dolphinscheduler/blob/d3a77c68e62b0c471e25cd6a4bfbd2c561dd2023/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java#L657-L659
> 
> .
> when exception throw in L658
> 
> https://github.com/apache/dolphinscheduler/blob/d3a77c68e62b0c471e25cd6a4bfbd2c561dd2023/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java#L658
> 
> 
> controller will catch it and package into `Result(code=Status.PROCESS_INSTANCE_STATE_OPERATION_ERROR.getCode(), message=Status.PROCESS_INSTANCE_STATE_OPERATION_ERROR.getMsg())`

Ok, I probably get it. I will take this approach to regulate the api module logging and exceptions in the new pr. Also, I will describe this specification in the logging specification doc.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java [hunk 4]
```diff
@@ -538,6 +549,18 @@ public Result queryProjectCreatedAndAuthorizedByUser(User loginUser) {
...
     }
 
+    /**
+     * check whether have read permission
+     *
+     * @param user user
+     * @param project project
+     * @return true if the user have permission to see the project, otherwise return false
+     */
+    private boolean checkReadPermission(User user, Project project) {
```
Comment (wen-hemin) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java:559]: Please confirm whether it is referenced

**Review by wen-hemin — COMMENTED** [review]
Please confirm.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java (hunk 4 above)
Comment (guowei-su) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java:559]: Sorry, this method was not referenced and was incorrectly merged while I was handling the conflict.

**Pull request merged**